ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFFromUtcTimestamp.java

```diff
@@ -88,18 +90,15 @@ public Object evaluate(DeferredObject[] arguments) throws HiveException {
     Timestamp inputTs = ((TimestampWritableV2) converted_o0).getTimestamp();
 
     String tzStr = textConverter.convert(o1).toString();
-    TimeZone timezone = TimeZone.getTimeZone(tzStr);
-
-    TimeZone fromTz;
-    TimeZone toTz;
-    if (invert()) {
-      fromTz = timezone;
-      toTz = tzUTC;
-    } else {
-      fromTz = tzUTC;
-      toTz = timezone;
+    if(ZoneInfo.getTimeZone(tzStr)==null){
```

Comment:
It would be nice if we added spaces between parentheses/operators.

```
if (ZoneInfo.getTimeZone(tzStr) == null) {
```

Reply:
Fixed.